lib/rouge/lexers/swift.rb

```diff
@@ -73,8 +75,8 @@ class deinit enum convenience extension final func import init internal lazy let
         rule /'(\\.|.)'/, Str::Char
         rule /(\d+\*|\d*\.\d+)(e[+-]?[0-9]+)?/i, Num::Float
```

Comment:
This rule as written doesn't match floating-point numbers with groupings (you can see this in the visual sample). I had a quick play around and I think this works:

```ruby
rule /(\d+(?:_\d+)*\*|(?:\d+(?:_\d+)*)*\.\d+(?:_\d)*)(e[+-]?\d+(?:_\d)*)?/i, Num::Float
```